ethereumkit/src/main/java/org/p2p/ethereumkit/external/core/GsonProvider.kt

```diff
@@ -0,0 +1,53 @@
+package org.p2p.ethereumkit.external.core
+
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import com.google.gson.reflect.TypeToken
+import org.p2p.ethereumkit.internal.api.jsonrpc.models.RpcBlock
+import org.p2p.ethereumkit.internal.api.jsonrpc.models.RpcTransaction
+import org.p2p.ethereumkit.internal.api.jsonrpc.models.RpcTransactionReceipt
+import org.p2p.ethereumkit.internal.models.DefaultBlockParameter
+import org.p2p.ethereumkit.internal.models.EthAddress
+import org.p2p.ethereumkit.internal.network.AddressTypeAdapter
+import org.p2p.ethereumkit.internal.network.BigIntegerTypeAdapter
+import org.p2p.ethereumkit.internal.network.ByteArrayTypeAdapter
+import org.p2p.ethereumkit.internal.network.DefaultBlockParameterTypeAdapter
+import org.p2p.ethereumkit.internal.network.IntTypeAdapter
+import org.p2p.ethereumkit.internal.network.LongTypeAdapter
+import org.p2p.ethereumkit.internal.network.OptionalTypeAdapter
+import java.math.BigInteger
+import java.util.Optional
+
+class GsonProvider {
+
+    var gson: Gson? = null
+
+    fun provide(): Gson {
+        return gson ?: GsonBuilder().also {
```

Comment:
apply would be good

also you can extract this also to a method and everything will look like this

fun provide(): Gson = gson ?: buildGson()